smarts/sstudio/genhistories.py

```diff
@@ -406,7 +406,9 @@ def _transform_all_data(self):
 
         map_width = self._dataset_spec["map_net"].get("width")
         if map_width:
-            valid_x = (df["position_x"] * self.scale).between(0, map_width)
+            valid_x = (df["position_x"] * self.scale).between(
+                df["length"] / 2, map_width - df["length"] / 2
+            )
```

Comment:
What is the intention with this change?

Reply:
I made that one.  This is to prevent agent vehicles from immediately being flagged as being off the map ("off-road" or "on-shoulder") when they first appear.  Each vehicle is  a different size in the NGSIM dataset, and they are coming from outside of the map in the "real world".   The first data-point we get for them is when their bbox centroid crosses into the map box.  But at that point in time, the vehicle's rear-end may not have made it onto the map yet, so the back tires will be marked as off road.

I originally made this change because I wanted to test the modified maps to see if vehicles were flagged for driving in holes and I was seeing "false positives" based on the map edges.  I only realized later that the map holes  would not cause off-road events anyway, but I think this change should remain (in particular because the imitation learning group uses our Events as feedback in the model training and testing).